### PR TITLE
Added methods

### PR DESCRIPTION
**Pull request by woolgathering**
I added a few methods for convenience and new functionality.

__zero__: a method to easily create zero vectors of arbitrary size
__clip__: clip a vector (distinct from limit)
__isVector__: returns true if it's a Vector (AbstractVector)
__rotate__: part of RealVector2D to rotate a vector

### CODE REVIEW

**Comment by woolgathering** [comment]
Just another ping... These changes are needed for [Boids](https://github.com/woolgathering/boids).

**Comment by nhthn** [comment]
dunno if this library is being actively maintained. cc @miguel-negrao

**Comment by miguel-negrao** [comment]
The commit has a lot of whitespace changes, what was the reason for this ? Difficult to see what was changed...

**Comment by woolgathering** [comment]
The earliest commit has a lot of whitespace because I didn't realize that the original class file used tab characters instead of spaces. I couldn't see it in my editor but saw it when I uploaded it to Github. Changed it back to tabs to be consistent later. This is the case with a lot of the whitespace (except some of the spaces just beyond the first curly braces; Atom removed them automatically on save...).

There were also some non-UTF8 characters that I deleted. Maybe I should have left those?

The actual changes:
- **zero()**: [RealVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L118), [RealVector2D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L195), and [RealVector3D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L228).
- **clip()**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L38)
- **isVector()**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L95)
- **rotate()**: [RealVector2D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L211)
- **azi()**: [RealVector3D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L253)
- **asRealVector3D**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L62)

Thoughts?

File: Vector.sc [hunk 1]
```diff
@@ -1,92 +1,106 @@
...
 	normalize {
 		^this / this.norm
 	}
-	
+
 	limit { |max|
 		if(this.norm > max, { ^this.normalize * max })
 	}
-	
+
```
Comment (miguel-negrao) [inline Vector.sc:37]: Indentation seems wrong here. Possibly not using tabs ?

File: Vector.sc [hunk 2]
```diff
@@ -1,92 +1,106 @@
...
 				if(j<k){
 					vector1 = vector1 - vector2.proj(vector1)
 				}
 			};
 			vector1.normalize;
-		}		
+		}
 	}
-		
+
```
Comment (miguel-negrao) [inline Vector.sc:94]: indentation
See comment at + Object

File: Vector.sc [hunk 3]
```diff
@@ -213,16 +250,20 @@ RealVector3D[slot] : RealVector {
 
 	phi { ^atan2(this[2], (this[0].squared + this[1].squared).sqrt) }
 
+	azi {
```
Comment (miguel-negrao) [inline Vector.sc:253]: Shouldn't azimuth be arctan( y / x ) ?

See [wikipedia](https://en.wikipedia.org/wiki/Spherical_coordinate_system#Coordinate_system_conversions)

File: object.sc [hunk 4]
```diff
@@ -0,0 +1,6 @@
++ Object {
```
Comment (miguel-negrao) [inline object.sc:1]: You can do o.isKindOf(AbstractVector). Is this really needed ?

File: Vector.sc [hunk 5]
```diff
@@ -1,92 +1,106 @@
-// �2009 Miguel Negr�o, Fredrik Olofsson
+// 2009 Miguel Negro, Fredrik Olofsson
```
Comment (miguel-negrao) [inline Vector.sc:1]: Please correct to 

    Copyright 2009 Miguel Negrão, Fredrik Olofsson

File: Fields.sc [hunk 6]
```diff
@@ -1,157 +1,157 @@
-// �2009 Miguel Negr�o
+// 2009 Miguel Negro
```
Comment (miguel-negrao) [inline Fields.sc:1]: Please change to "Copyright 2009 Miguel Negrão"

**Review by miguel-negrao — CHANGES_REQUESTED** [review]
Regarding the whitespace changes, trailing whitespace is fine (thanks !) but changes from tabs to spaces is not.

**Comment by miguel-negrao** [comment]
If your git skills are up to it, it would be nice to have a single commit just with the trailing whitespace and weird character removal changes, followed by a single commit with your actual changes.

**Comment by woolgathering** [comment]
No problem. I'll get to it this weekend!

**Comment by woolgathering** [comment]
I'll close this request and make a new one that's cleaner.

**Comment by miguel-negrao** [comment]
sorry, forgot the word "removing" in "Regarding the whitespace changes, REMOVING trailing whitespace is fine (thanks !) but changes from tabs to spaces is not."

File: Vector.sc (hunk 1 above)
Comment (woolgathering) [reply Vector.sc:37]: This is caused by the removal of trailing whitespace. Same thing happens all over the file.

File: Vector.sc (hunk 3 above)
Comment (woolgathering) [reply Vector.sc:253]: Yes, I can do it that way to keep things consistent. I just through of turning the entire 3D space on its side and doing the same operation about the x-axis as opposed to the z-axis.

File: object.sc (hunk 4 above)
Comment (woolgathering) [reply object.sc:1]: Not anymore. I removed that line in Boids and can remove it here, too.

File: Vector.sc (hunk 5 above)
Comment (woolgathering) [reply Vector.sc:1]: No problem.

File: Fields.sc (hunk 6 above)
Comment (woolgathering) [reply Fields.sc:1]: See above

**Pull request closed without merging**